### PR TITLE
feat: introduce statefulset

### PR DESCRIPTION
**Pull request by smuu**
Resolves https://github.com/celestiaorg/knuu/issues/82

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: pkg/k8s/k8s_statefulset.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	statefulset, err := Clientset().AppsV1().StatefulSets(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get statefulSet %s: %w", name, err)
+	}
+
+	return statefulset, nil
+}
+
+// DeployStatefulSet creates a new statefulSet in the given namespace if it doesn't already exist.
+func DeployStatefulSet(statefulSetConfig StatefulSetConfig, init bool) (*appv1.StatefulSet, error) {
```
Comment (jrmanes) [inline pkg/k8s/k8s_statefulset.go:32]: maybe we can call it `CreateStatefulSet`

Comment (jrmanes) [reply pkg/k8s/k8s_statefulset.go:32]: just to match the references to `create` within the function

**Review by jrmanes — APPROVED** [review]
just a comment, LGTM! legend! 👏 🔥

File: pkg/k8s/k8s_statefulset.go (hunk 1 above)
Comment (smuu) [reply pkg/k8s/k8s_statefulset.go:32]: Yeah, the `k8s` package is a mess currently. The naming could be more consistent, but I created an issue for that. See https://github.com/celestiaorg/knuu/issues/108 and https://github.com/celestiaorg/knuu/issues/111.

**Pull request merged**